From 723b6224251834d3c1dea8243dba444a4099ece5 Mon Sep 17 00:00:00 2001
From: Florian Schmaus <flo@geekplace.eu>
Date: Thu, 14 Nov 2019 13:39:49 +0100
Subject: [PATCH] setup.py: package_data must be a mapping from name to a list

The current setyp.py causes a TypeERror because it uses a tuple where
a list should be used:

running build_py
Traceback (most recent call last):
  File "setup.py", line 238, in <module>
    "tar": tar,
  File "/usr/lib64/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib64/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib64/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "setup.py", line 97, in run
    distutils.command.build.build.run (self)
  File "/usr/lib64/python3.6/distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
  File "/usr/lib64/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib64/python3.6/distutils/dist.py", line 973, in run_command
    cmd_obj.ensure_finalized()
  File "/usr/lib64/python3.6/distutils/cmd.py", line 107, in ensure_finalized
    self.finalize_options()
  File "/usr/lib64/python3.6/distutils/command/build_py.py", line 56, in finalize_options
    self.data_files = self.get_data_files()
  File "/usr/lib64/python3.6/distutils/command/build_py.py", line 116, in get_data_files
    file[plen:] for file in self.find_data_files(package, src_dir)
  File "/usr/lib64/python3.6/distutils/command/build_py.py", line 124, in find_data_files
    + self.package_data.get(package, []))
TypeError: can only concatenate list (not "tuple") to list

Fixes https://bugs.launchpad.net/rubber/+bug/1830640
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index e924cec..5515eee 100644
--- a/setup.py
+++ b/setup.py
@@ -224,7 +224,7 @@ Metapost compilation).\
         "rubber" : "src",
     },
     package_data = {
-        "rubber" : ("rules.ini", ),
+        "rubber" : ["rules.ini"],
     },
     scripts = (
         "rubber",
-- 
2.23.0

